### PR TITLE
feat: Add endpoints for search actions

### PR DESCRIPTION
**Pull request by AmanuelAaron**
## Ticket

ET-706
ET-707
ET-708
ET-710
ET-711
ET-712
ET-713
ET-714
ET-715
ET-716
ET-717
ET-718
ET-719
ET-721

## Description

Adding archive, unarchive, pause, resume, delete, kill with bulk action

## Test Plan

Existing tests should pass

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_search.go [hunk 1]
```diff
@@ -169,3 +169,230 @@ func (a *apiServer) DeleteSearchTag(
 	}
 	return &res, nil
 }
+
+func BulkFiltersSearchtToExperiment(filter *apiv2.BulkSearchFilters) *apiv1.BulkExperimentFilters {
```
Comment (jesse-amano-hpe) [inline master/internal/api_search.go:173]: Small typo here -- hopefully a tool can help rename it (F2 in VSCode, or find+replace would probably do it in this case)

Comment (AmanuelAaron) [reply master/internal/api_search.go]: Fixed

File: master/internal/api_search.go [hunk 2]
```diff
@@ -214,3 +214,230 @@ func (a *apiServer) PutSearchRetainLogs(
...
+	}
+	_, err := a.DeleteExperiment(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.DeleteSearchResponse{}
+	return &res, nil
+}
+
+func (a *apiServer) DeleteSearches(ctx context.Context, req *apiv2.DeleteSearchesRequest,
```
Comment (carolinaecalderon) [inline master/internal/api_search.go:428]: tests?

**Review by carolinaecalderon — COMMENTED** [review]
looks good, needs tests though

**Review by jesse-amano-hpe — APPROVED** [review]

File: master/internal/api_search.go (hunk 2 above)
Comment (AmanuelAaron) [reply master/internal/api_search.go:428]: These endpoints are essentially wrappers around already tested endpoints. Added a test for the newly added bulk conversion filter function

**Review by carolinaecalderon — APPROVED** [review]

**Pull request merged**